src/prefect/core/task.py

```diff
@@ -262,6 +265,15 @@ def __init__(
         else:
             self.result = result
 
+        self.target = target
+
+        if result and target:
```

Comment:
we only need to warn if `result.location` is actually set, otherwise it's all good
```suggestion
        if result.location and target:
```